infra/src/main/kotlin/fi/fta/geoviite/infra/inframodel/InfraModelService.kt

```diff
@@ -112,6 +121,18 @@ constructor(
         return tryParsing(overrideParameters?.source) { validateInternal(file, overrideParameters) }
     }
 
+    fun getInfraModelsZipped(ids: List<IntId<GeometryPlan>>, translation: Translation): ByteArray =
+        geometryService
+            .getPlanHeaders { header -> ids.contains(header.id) }
+            .map { header -> Pair(header, geometryService.getPlanFile(header.id, translation)) }
+            .distinctBy { it.first.fileName }
```

Comment:
Meillä voi olla kannassa samannimisiä suunnitelmatiedostoja. En oikein tiedä miten tämä kuuluisi hanskata ns. oikeaoppisesti... tämä on yksi tapa, toinen olisi muokata duplikaattien tiedostonimiä (esim. lisätä joku `(1)` ennen päätettä. Silloin tosin menee vaikeammaksi tulkita mikä tiedosto oikeasti vastasi mitäkin suunnitelmaa Geoviitteessä. Vähän damned if you do, damned if you don't 🤷